docs/spec/tuples.rst

```diff
@@ -0,0 +1,155 @@
+Tuples
+======
+
+The ``tuple`` class has some special behaviors and properties that make it
+different from other classes from a typing perspective. The most obvious
+difference is that ``tuple`` is variadic -- it supports an arbitrary number
+of type arguments. At runtime, the sequence of objects contained within the
+tuple is fixed at the time of construction. Elements cannot be added, removed,
+reordered, or replaced after construction. These properties affect subtyping
+rules and other behaviors as described below.
+
+
+Tuple Type Form
+---------------
+
+The type of a tuple can be expressed by listing the element types. For
+example, ``tuple[int, int, str]`` is a tuple containing an ``int``, another
+``int``, and a ``str``.
+
+The empty tuple can be annotated as ``tuple[()]``.
+
+Arbitrary-length homogeneous tuples can be expressed using one type and
+ellipsis, for example ``tuple[int, ...]``. This type is equivalent to a union
+of tuples containing zero or more ``int`` elements (``tuple[()] |
+tuple[int] | tuple[int, int] | tuple[int, int, int] | ...``).
+Arbitrary-length homogeneous tuples are sometimes referred to as "unbounded
+tuples". Both of these terms appear within the typing spec, and they refer to
+the same concept.
+
+The type ``tuple[Any, ...]`` is special in that it is bidirectionally
```

Comment:
What makes this case special? Doesn't this follow from consistent subtyping rules for any and the equivalence you gave above?

Reply:
This is special because it's a gradual type with respect to the length of the tuple. It is bidirectionally compatible with any tuple. The use of `tuple[T, ...]` where `T` is any type other than `Any` is _not_ a gradual type. It is shorthand for a union type. A value with this type is not assignable to a tuple with a known fixed length. The next section contains some specific examples that help demonstrate this difference.

Reply:
Thanks for the clarification, Eric! I assume `T` does not have to be exactly `Any` to `tuple[T, ...]` to be a gradual type. It just needs to contain `Any`, right?

Reply:
> I assume T does not have to be exactly Any to tuple[T, ...] to be a gradual type.

No, it needs to be exactly `Any`. I think the text and examples are already pretty clear on this account, but if you can think of a way to make it clearer, let me know.